### DISCUSSION TITLE
How to enable comma-seperated numbers as form input?

### DISCUSSION THREAD

**Discussion by Arturexe** [google/filament#6140] [Q&A] [answered]
I'm saving currency values as integer in my database and want the user to be able to enter a `comma-seperated float value` in the form, because that's common practice in germany. It works with `dot-seperated` values, but as soon as I enter a comma value, the form responds with `The price must be a number value.`.

I've tried mutating the value before sending it to the database in `CreateProduct.php`:
```
protected function mutateFormDataBeforeCreate(array $data): array
{
    $data['price'] = floatval(str_replace(',', '.', $data['price']));
    $data['price'] = $data['price'] * 100;
    
    return $data;
}
```

but this code part does not even get executed before the error is shown. I've tested with `dd()`, which is not shown even if it is placed at the beginning of the function.

**Comment by pixelflinger** [comment]
Probably not the right place.

**Comment by Arturexe** [comment, marked as answer]
I had a wrong approach, using `->rule('numeric')` during validation instead of `->numeric()` function, which allows for both delimiters to be used.

**Reply by bejado** [reply]
I think you want the https://github.com/filamentphp/filament repo
